src/coreclr/jit/importercalls.cpp

```diff
@@ -3093,6 +3094,29 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Type_get_TypeHandle:
+            {
+                assert(IsTargetAbi(CORINFO_NATIVEAOT_ABI));
+
+                // Try to avoid ClsHandle -> Type object -> ClsHandle roundtrip:
+                GenTree* op1 = impStackTop(0).val;
+                if (op1->IsHelperCall() && gtIsTypeHandleToRuntimeTypeHelper(op1->AsCall()))
+                {
+                    // struct RuntimeTypeHandle { IntPtr _value; }
+                    assert(info.compCompHnd->getClassNumInstanceFields(sig->retTypeClass) == 1);
+
+                    unsigned structLcl = lvaGrabTemp(true DEBUGARG("RuntimeTypeHandle"));
+                    lvaSetStruct(structLcl, sig->retTypeClass, false);
+                    GenTree*       realHandle   = op1->AsCall()->gtArgs.GetUserArgByIndex(0)->GetNode();
+                    GenTreeLclFld* handleFld    = gtNewLclFldNode(structLcl, realHandle->TypeGet(), 0);
+                    GenTree*       asgHandleFld = gtNewAssignNode(handleFld, realHandle);
```

Comment:
Is it required for this to create a clone of the op1 first arg? I would expect that we just strip the helper call and return the the op1 first arg without cloning it.